### ISSUE TITLE
don't forget privileges in the new syncmanager and syncmanager api

### ISSUE THREAD

**Issue by urapadmin** [urapadmin/kiosk#992]

**Comment by urapadmin** [comment]
- [ ] sync manager routes
- [ ] KioskFileMakerWorkstation routes
- [ ] KioskFileMakerWorkstation menus
- [ ] sync manager api routes
- [ ] sync manager menus

**Comment by urapadmin** [comment]
### @test 0.11
- [ ] only users with privilege "edit workstation" should see the menu item "reset all" in the field manager
- [ ]

**Comment by urapadmin** [comment]
### @test 0.11 (for developer)
#### api
- [ ] DELETE:/api/syncmanager/v1/workstation/{ws_id}/job needs "edit workstation" privilege

**Comment by lbestock** [comment]
(Side note: it does not work to test privileges with two browser windows of the same browser - logging in as one user in one changes the login in the other.)

**Comment by urapadmin** [comment]
yes, and that is correct behavior.

**Comment by lbestock** [comment]
As far as I have seen so far once logged in in different browsers, everything is working properly. But I will have to go through it yet again when not tired. And a different UI for privileges would be a desideratum if we were thrown a million dollars; even here in the ticket and in the privileges as on the wiki things are not called precisely what they are called in kiosk (eg "create workstation" vs "new workstation"), and the inability to see what options exist when one is defining things is a problem for non-developer users.

**Comment by lbestock** [comment]
I think I broke another user, or at least I am having more privilege problems. 
I could not test that upload and download privileges were separate. I was testing with Test3, who was only a field worker. I was directly addressing additional privileges to him (like "edit workstation") in kiosk privileges and things were changing as expected. But a fieldworker has both upload and download permissions. So I wanted to sever the user from a group that had privileges. I discovered that I could not assign him to no group; I was able to login still but got this: 
![image](https://user-images.githubusercontent.com/49291909/135124603-93dcdf92-da43-45a1-9899-7df652209228.png)
I then tried to assign him to group user or users (tried both) as I thought that might do it, but I go the same message. 
I then made him a field worker again but went into privileges and removed all of the privileges addressed to field_worker except modify data. But I still get that message even though I am in a group that I know exists and has at least one defined privilege.
I defined another privilege for Test3: prepare workstation. But that hasn't done it, I am still not able to get in as Test3. Only now I'm not able to get in as Test2, either - same message, though Test2 is both a field_worker and an operator and I didn't mess with privileges for operators at all. 
I am able to get in properly as myself.
I will leave this. I am in danger of creating too much of a muddle and making it too hard for you to see what I have done and if it means anything at all.

**Comment by urapadmin** [comment]
you already made a muddle. You need a user's manual on privileges. They are not, as I am sure some TA would put it, "intuitive". Let me try to explain why your attempts failed step by step:

- If you have a user without a group you must assign all privileges to that user that are necessary to do anything at all individually. Since PVD is configured to load the field manager after log in, the user must have the most basic privileges to open that one. It is "download workstation". That basic information would have been useful for you. I'll add it to the wiki. A user without any privileges, like the one you seem to have created, can enter the file repository, though. But not the field manager. A system where it is possible that users don't have the basic privilege for the field manager would have to be configured to load the file repository after log in. 

- you assigned the user to a group user or users? But such a group does not exist unless there are privileges assigned to a group. In fact it is the first privilege assigned to a group that makes it a group. I can't see any "user" or "users" on pvd. 

- removing all privileges from field_worker is really tampering with all the other users using "field_worker". That can make the life of other testers unexpectedly hard.

- "modify data" is a file repository privilege, I I remember correctly, so that would not help with the field manager. 

- the group "operator" does not add the "download workstation" privilege. It adds privileges for operators. A user who is an operator would not be just in group "operator", an operator would be in group "field_worker" and "operator". That is at least how the privileges in PVD are organized. So removing the privileges from field_worker was not compensated by adding a user to operator. Now you really need to give the field_worker group back their privileges!

So as far as I can see all is fine. If you want to test what you wanted to test, this is what you should do:
- create a user test3 and assign the "download workstation" privilege to it. You should be able to enter the field manager with that user and download a workstation. But you should not be able to upload a workstation.
- add the "upload workstation" privilege to the individual user and you can do that, too.

**Comment by urapadmin** [comment]
> 
> 
> As far as I have seen so far once logged in in different browsers, everything is working properly. But I will have to go through it yet again when not tired. And a different UI for privileges would be a desideratum if we were thrown a million dollars; even here in the ticket and in the privileges as on the wiki things are not called precisely what they are called in kiosk (eg "create workstation" vs "new workstation"), and the inability to see what options exist when one is defining things is a problem for non-developer users.

If you found a deviation between wiki and software, have you fixed it in the wiki?

**Comment by lbestock** [comment]
See above where I said I should test this ticket only when not tired? Well..
I think I have undone my muddle, wildtester. I had a screen shot of the privileges I deleted from field_worker, so I could restore them (I knew that needed to be very temporary). I will go back and look at the wiki now, too, and make sure things are the same, but I suspect I wasn't quite understanding some of it before. This muddle has at least resulted in me getting privileges better.

**Comment by lbestock** [comment]
"users" as a group was suggested to me by the browser, and as is my want I just followed that path without thinking what else would have to be checked to see if the path made sense.

**Comment by lbestock** [comment]
What I have not learned is how to spell privileges, which trips me up every single time.
But this test is done, including with the extra special bit in the wiki but not here of trying to get into admin from a pleb user account by typing in the address in the browser, and not being allowed. I did not find any inconsistencies in the wiki, which is extremely thorough on the privileges (no d, I always want a d in it) so I don't know what I was talking about in that case. And privileges are cooler than I realized.
